docker/Dockerfile

```diff
@@ -1,25 +1,28 @@
-FROM ubuntu:xenial
+FROM debian:stretch-slim
 
-RUN apt-get update -y && apt-get upgrade -y && apt-get install -y \
-build-essential \
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+g++ \
+make \
 cmake \
 libboost-dev \
 libboost-filesystem-dev \
 libboost-program-options-dev \
 libboost-regex-dev \
 libsqlite3-dev \
 ragel \
-wget
+&& apt-get clean \
```

Comment:
Adding a comment like `Clean up to reduce layer size` before this line would be useful to make a clear separation of the actions.